### PR TITLE
[LiveLeak] Check if the original videos exist (closes #21206)

### LINKED ISSUE #21206: LiveLeak: some videos return HTTP Error 403
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2019.05.20**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.liveleak.com/view?t=C26ZZ_1558612804']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.05.20
[debug] Git HEAD: 25b83c2a0
[debug] Python version 3.7.3 (CPython) - Linux-5.1.4-arch1-1-ARCH-x86_64-with-arch
[debug] exe versions: ffmpeg 4.1.3, ffprobe 4.1.3, phantomjs ., rtmpdump 2.4
[debug] Proxy map: {}
[LiveLeak] C26ZZ_1558612804: Downloading webpage
[download] Downloading playlist: Liveleak.com - Stray dog nearly wins 100m race after joining in accidentally
[LiveLeak] playlist Liveleak.com - Stray dog nearly wins 100m race after joining in accidentally: Collected 1 video ids (downloading 1 of them)
[download] Downloading video 1 of 1
[debug] Default format spec: bestvideo+bestaudio/best
[debug] Invoking downloader on 'https://cdn.liveleak.com/80281E/ll_a_s/2019/May/23/LiveLeak-dot-com-Ih8Y_1558612804-LbnxSMi8jWXVzuYSWml0tgjM2nnK7dj21558611813mp4.mp4?Wk0GS8abpvNJ7QeS3f7boxotGr6iGqNlOYpfW0GbwCV5FbSzf8lZqdJIvBw5tzmdB6r8Hc8pV3eAW9wqT1aVzHadkMVngpbTEOIpaaKCZ8b8IWdQP8Ku4rNon9CkOLY7cc6_J9MZio86w7h5lgTxAw'
ERROR: unable to download video data: HTTP Error 403: Forbidden
Traceback (most recent call last):
  File "/home/bitraid/src/youtube-dl/youtube_dl/YoutubeDL.py", line 1915, in process_info
    success = dl(filename, info_dict)
  File "/home/bitraid/src/youtube-dl/youtube_dl/YoutubeDL.py", line 1854, in dl
    return fd.download(name, info)
  File "/home/bitraid/src/youtube-dl/youtube_dl/downloader/common.py", line 364, in download
    return self.real_download(filename, info_dict)
  File "/home/bitraid/src/youtube-dl/youtube_dl/downloader/http.py", line 341, in real_download
    establish_connection()
  File "/home/bitraid/src/youtube-dl/youtube_dl/downloader/http.py", line 109, in establish_connection
    ctx.data = self.ydl.urlopen(request)
  File "/home/bitraid/src/youtube-dl/youtube_dl/YoutubeDL.py", line 2227, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python3.7/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib/python3.7/urllib/request.py", line 641, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python3.7/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.7/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

## Description

Started with #4768, the extracted LiveLeak urls are modified (remove `.*.mp4`) to include the "original" version(s) of the video (without the "Liveleak" logo) and also use it as "best". It seems that lately there are some videos that don't have an "original" version, and while the extracted urls work fine - the modified ("original") ones result in: `HTTP Error 403: Forbidden`. These videos also don't have the "LiveLeak" logo, but have the "newsflare" logo. There doesn't seem to be any obvious difference in the urls of the videos that doesn't have an "original" version from those that do, so i'm not sure what the best fix would be.

Only the first two formats (from the unmodified urls) work for this video:
```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-F', '-v', 'https://www.liveleak.com/view?t=C26ZZ_1558612804']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.05.20
[debug] Git HEAD: 25b83c2a0
[debug] Python version 3.7.3 (CPython) - Linux-5.1.4-arch1-1-ARCH-x86_64-with-arch
[debug] exe versions: ffmpeg 4.1.3, ffprobe 4.1.3, phantomjs ., rtmpdump 2.4
[debug] Proxy map: {}
[LiveLeak] C26ZZ_1558612804: Downloading webpage
[download] Downloading playlist: Liveleak.com - Stray dog nearly wins 100m race after joining in accidentally
[LiveLeak] playlist Liveleak.com - Stray dog nearly wins 100m race after joining in accidentally: Collected 1 video ids (downloading 1 of them)
[download] Downloading video 1 of 1
[info] Available formats for C26ZZ_1558612804:
format code    extension  resolution note
360p           mp4        360p       
720p           mp4        720p       
original-360p  mp4        unknown    
original-720p  mp4        unknown    (best)
[download] Finished downloading playlist: Liveleak.com - Stray dog nearly wins 100m race after joining in accidentally
```

Their urls doesn't seem to have any differences from the other videos:

- `Error 403` when removing `.*.mp4` (`.5ce68b68c42ea.mp4`) from:
`https://cdn.liveleak.com/80281E/ll_a_s/2019/May/23/LiveLeak-dot-com-Ih8Y_1558612804-LbnxSMi8jWXVzuYSWml0tgjM2nnK7dj21558611813mp4.mp4.5ce68b68c42ea.mp4?2uCjJmr004dtCQ-BFJlaAuCczOd_rc1bk3FD-_3r6IxzZjgXgmSxIWCHl9jfHWvAHLrY8c_B0957_9Sql75Tq9tWKnmUQfP6un0USD410MVIy1M9d2mbh2g88XaU8IYe-hhdAoDY_epUHyKc2zb32g`

- Gets the "original" video when removing `.*.mp4` (`.5ce66383ad172.mp4`) from:
`https://cdn.liveleak.com/80281E/ll_a_s/2019/May/23/LiveLeak-dot-com-149530jakartaaa_1558602606.mp4.5ce66383ad172.mp4?1DvRuMj1HtdJBRz7h-2990qTokZQjSNvFjLAqOvG3Uz-PTStbyzYKfVGKcacqRx1VYGK80Tc2u6Iox6R2D3fhHb6PvNzT6fvjrmNET1ycDwzHpWFIsSpnc44fZ8BTswDTbruQs__OdkuyCkRMeNxXw`

### PR DESCRIPTION
**Pull request by bitraid**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

A small addition for fixing #21206: Check ~~for HTTPErrors~~ `_is_valid_url` before accepting the stripped urls.

### CODE REVIEW

File: youtube_dl/extractor/liveleak.py [hunk 1]
```diff
@@ -132,6 +137,12 @@ def _real_extract(self, url):
                 # the same video without the LiveLeak logo at the top (see
                 # https://github.com/ytdl-org/youtube-dl/pull/4768)
                 orig_url = re.sub(r'\.mp4\.[^.]+', '', a_format['url'])
+                # Check if the raw video exists
+                try:
+                    compat_urllib_request.urlopen(orig_url).read()
```
Comment (remitamine) [inline youtube_dl/extractor/liveleak.py]: use `InfoExtractor._is_valid_url`.

Comment (bitraid) [reply youtube_dl/extractor/liveleak.py]: Ok, done.

File: youtube_dl/extractor/liveleak.py [hunk 2]
```diff
@@ -134,11 +138,12 @@ def _real_extract(self, url):
                 orig_url = re.sub(r'\.mp4\.[^.]+', '', a_format['url'])
                 if a_format['url'] != orig_url:
                     format_id = a_format.get('format_id')
-                    formats.append({
-                        'format_id': 'original' + ('-' + format_id if format_id else ''),
-                        'url': orig_url,
-                        'preference': 1,
-                    })
+                    if self._is_valid_url(orig_url, video_id, format_id):
```
Comment (remitamine) [inline youtube_dl/extractor/liveleak.py]: the message will be `Checking None URL` if `format_id` is not available, provide a fallback for `format_id`.

Comment (bitraid) [reply youtube_dl/extractor/liveleak.py]: Is 9ac2735 ok?

**Comment by remitamine** [comment]
Thanks.

**Pull request merged**